Pearson-Advance/course-discovery
# PADV 356 - Integrate the build process with Discovery

## Ticket

https://agile-jira.pearson.com/browse/PADV-356

## Description

This PR aims to add the workflow for automatically building the discovery tutor image.

## Type of Change

- [x] Add a new workflow for image building
- [x] Including the tutor-image-building-action action in the workflow
- [x] Trigger workflow on merge with main and stage branches

## How to test

- Create a test PR in the discovery repository.
- Merge this PR into the target branch.
- Check that the Github Action is triggered by the PR merge event.
- Check the output of the image build creation for any issues or silent issues.
-  When the process is complete, check the Docker Hub Pearson repository and verify that the image has been built (https://hub.docker.com/repository/docker/paops/discovery-tutor/general).
-  Use the image in a local environment to test it out.
- You can use this repo https://github.com/Pearson-Advance/discovery/tree/vue/PADV-356.tests to perform any tests

## Reviewers
